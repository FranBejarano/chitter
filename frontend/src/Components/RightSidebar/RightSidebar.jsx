import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import { logout } from "../../redux/userSlice";
import Signin from "../Signin/Signin";

const RightSidebar = () => {

  const { currentUser } = useSelector((state) => state.user);
  const dispatch = useDispatch();

  const handleLogout = () => {
    dispatch(logout());
  };

  return (
    <>
      {!currentUser ?
        (<Signin />
        ) : (
          <Link to="/">
            <button
              className="bg-red-500 px-4 py-2 text-white rounded-full"
              onClick={handleLogout}
            >
              Logout
            </button>
          </Link>
        )}
    </>
  );
};

export default RightSidebar;
