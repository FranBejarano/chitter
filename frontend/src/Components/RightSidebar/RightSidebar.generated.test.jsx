import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';
import RightSidebar from "./RightSidebar";

jest.mock("react-redux");
jest.mock("react-router-dom");
jest.mock("../../redux/userSlice");
jest.mock("../Signin/Signin");

const renderTree = tree => renderer.create(tree);
describe('<RightSidebar>', () => {
  it('should render component', () => {
    expect(renderTree(<MemoryRouter><RightSidebar
    /></MemoryRouter>).toJSON()).toMatchSnapshot();
  });

});