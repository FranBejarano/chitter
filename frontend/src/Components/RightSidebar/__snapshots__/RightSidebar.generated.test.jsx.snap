// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RightSidebar> should render component 1`] = `null`;
